python_modules/libraries/dagster-airlift/dagster_airlift/core/load_defs.py

```diff
@@ -326,3 +331,42 @@ def load_airflow_dag_asset_specs(
         retrieval_filter=retrieval_filter or AirflowFilter(),
     ).get_or_fetch_state()
     return list(spec_iterator(construct_dag_assets_defs(serialized_data)))
+
+
+def uri_to_asset_key(uri: str) -> AssetKey:
+    last_path_segment = uri.split("/")[-1]
+    with_ext_removed = last_path_segment.split(".")[0]
+    return AssetKey(with_ext_removed)
+
+
+def construct_dataset_defs(
+    serialized_data: SerializedAirflowDefinitionsData,
+) -> Sequence[AssetsDefinition]:
+    """Construct dataset definitions from the serialized Airflow data."""
+    from dagster_airlift.core.multiple_tasks import assets_with_multiple_task_mappings
+
+    return cast(
+        "Sequence[AssetsDefinition]",
+        [
+            assets_with_multiple_task_mappings(
+                task_handles=[
+                    {"dag_id": t.dag_id, "task_id": t.task_id} for t in dataset.producing_tasks
+                ],
+                assets=[
+                    external_asset_from_spec(
+                        AssetSpec(
+                            key=uri_to_asset_key(dataset.uri),
+                            metadata=dataset.extra,
```

Comment:
should we toss the original uri in the metadata as well?

Reply:
yea almost certainly